service/lib/agama/storage/proposal_settings_conversion/to_y2storage.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage"
+require "agama/storage/volume_conversion"
+require "agama/storage/volume_templates_builder"
+
+module Agama
+  module Storage
+    module ProposalSettingsConversion
+      # Proposal settings conversion to Y2Storage format.
+      class ToY2Storage
+        # @param settings [Agama::Storage::ProposalSettings]
+        # @param config [Agama::Config]
+        def initialize(settings, config:)
+          @settings = settings
+          @config = config
+        end
+
+        # Performs the conversion to Y2Storage format.
+        #
+        # @return [Y2Storage::ProposalSettings]
+        def convert
+          # Despite the "current_product" part in the name of the constructor, it only applies
+          # generic default values that are independent of the product (there is no YaST
+          # ProductFeatures mechanism in place).
+          Y2Storage::ProposalSettings.new_for_current_product.tap do |target|
+            boot_device_conversion(target)
+            candidate_devices_conversion(target)
+            lvm_conversion(target)
+            encryption_conversion(target)
+            space_policy_conversion(target)
+            volumes_conversion(target)
+          end
+        end
+
+      private
+
+        # @return [Agama::Storage::ProposalSettings]
+        attr_reader :settings
+
+        # @return [Agama::Config]
+        attr_reader :config
+
+        # @param target [Y2Storage::ProposalSettings]
+        def boot_device_conversion(target)
+          target.root_device = settings.boot_device
+        end
+
+        # @param target [Y2Storage::ProposalSettings]
+        def candidate_devices_conversion(target)
+          candidate_devices = []
+          candidate_devices = settings.lvm.system_vg_devices if settings.lvm.enabled?
+          candidate_devices = [settings.boot_device] if candidate_devices.none?
```

Comment:
well, more common is to use `empty?` instead of `none?`.

Reply:
ah, it is used here to filter if there are only nils in array....definitively tricky usage. Maybe more readable will be using compact already in a line above or here in test and use common `empty?` as I think it can confuse more people.

Reply:
well, I think that code you write and code above looks quite different. So let me write your code from comment in consistent way:
```
candidate_devices = if settings.lvm.enabled? && settings.lvm.system_vg_devices.any?
  settings.lvm.system_vg_devices
else
  [settings.boot_device]
end.compact
```

this way it will be at least visible that candidates are either system_vg_devices or boot_device and not some strange combination of both.